src/ccl_background.c

```diff
@@ -131,29 +131,51 @@ double ccl_rho_x(ccl_cosmology * cosmo, double a, ccl_species_x_label label, int
 }
 
 /* --------- ROUTINE: ccl_mu_MG ---------
-INPUT: cosmology object, scale factor
-TASK: Compute mu(a) where mu is one of the the parameterizating functions
+INPUT: cosmology object, scale factor, wavenumber for scale 
+TASK: Compute mu(a,k) where mu is one of the the parameterizating functions
 of modifications to GR in the quasistatic approximation.
 */
 
-double ccl_mu_MG(ccl_cosmology * cosmo, double a, int *status)
+double ccl_mu_MG(ccl_cosmology * cosmo, double a, double k, int *status)
 {
-    // This function can be extended to include other
-    // z-dependences for mu in the future.
-    return cosmo->params.mu_0 * ccl_omega_x(cosmo, a, ccl_species_l_label, status) / cosmo->params.Omega_l;
+    double s1_k, s2_k, hnorm;
+	// This function can be extended to include other
+	// redshift and scale z-dependences for mu in the future
+	hnorm = h_over_h0(a, cosmo, status);
+    if (k==0.0) {
+        s1_k = cosmo->params.c1_mg;
+    }
+    else {
+	    s2_k = (cosmo->params.lambda_mg*(hnorm*cosmo->params.H0)/k/(ccl_constants.CLIGHT/1000));
+	    s1_k = (1.0+cosmo->params.c1_mg*s2_k*s2_k)/(1.0+s2_k*s2_k);
+	}
+	return cosmo->params.mu_0 * ccl_omega_x(cosmo, a, ccl_species_l_label, status)/cosmo->params.Omega_l*s1_k;
 }
 
 /* --------- ROUTINE: ccl_Sig_MG ---------
-INPUT: cosmology object, scale factor
-TASK: Compute Sigma(a) where Sigma is one of the the parameterizating functions
+INPUT: cosmology object, scale factor, wavenumber for scale
+TASK: Compute Sigma(a,k) where Sigma is one of the the parameterizating functions
 of modifications to GR in the quasistatic approximation.
 */
 
-double ccl_Sig_MG(ccl_cosmology * cosmo, double a, int *status)
+double ccl_Sig_MG(ccl_cosmology * cosmo, double a, double k, int *status)
 {
-    // This function can be extended to include other
-    // z-dependences for Sigma in the future.
-    return cosmo->params.sigma_0 * ccl_omega_x(cosmo, a, ccl_species_l_label, status) / cosmo->params.Omega_l;
+    double s1_k, s2_k, hnorm;
+	// This function can be extended to include other
+	// redshift and scale dependences for Sigma in the future.
+	hnorm = h_over_h0(a, cosmo, status);
+    if (k==0.0) {
+        s1_k = cosmo->params.c2_mg;
+    }
+    else {
+	    s2_k = cosmo->params.lambda_mg*(hnorm*cosmo->params.H0)/k/(ccl_constants.CLIGHT/1000);
+        s1_k = (1.0+cosmo->params.c2_mg*s2_k*s2_k)/(1.0+s2_k*s2_k);
+
+//		printf("c1=%f,c2=%f,lambda=%f \n",cosmo->params.c1_mg,cosmo->params.c2_mg,cosmo->params.lambda_mg);
```

Comment:
Should clean up these print statements at some point.

Reply:
Done.